### PR TITLE
[feature](bitwise function) bit_count/bit_shift_left/bit_shift_right implementation

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
1. bit_count(x): return the exist count of one in 2's complement represent of integer x.
2. bit_shift_left(x, c): do logical left shift to x by c bits, and return result as a BIGINT.
3. bit_shift_right(x, c): do logical right shift to x by c bits, and return result as a BIGINT.

NOTE: bit_shift will return same type with x, so bit_shift_left(BIGINT_MAX, 1) will get -2

```sql
WITH my_cte AS (   SELECT 9223372036854775807 AS BIGINT_MAX ) SELECT BIGINT_MAX, bit_count(BIGINT_MAX) FROM my_cte
--------------

+---------------------+-----------------------+
| BIGINT_MAX          | bit_count(BIGINT_MAX) |
+---------------------+-----------------------+
| 9223372036854775807 |                    63 |
+---------------------+-----------------------+
1 row in set (0.02 sec)

WITH my_cte AS (   SELECT 9223372036854775807 AS BIGINT_MAX ) SELECT BIGINT_MAX, bit_shift_left(BIGINT_MAX, 1) FROM my_cte
--------------

+---------------------+-------------------------------+
| BIGINT_MAX          | bit_shift_left(BIGINT_MAX, 1) |
+---------------------+-------------------------------+
| 9223372036854775807 |                            -2 |
+---------------------+-------------------------------+
1 row in set (0.01 sec)

WITH my_cte AS (   SELECT 9223372036854775807 AS BIGINT_MAX ) SELECT BIGINT_MAX, bit_shift_right(BIGINT_MAX, 62) FROM my_cte
--------------

+---------------------+-----------------------------------+
| BIGINT_MAX          | bit_shift_right(`BIGINT_MAX`, 62) |
+---------------------+-----------------------------------+
| 9223372036854775807 |                                 1 |
+---------------------+-----------------------------------+
```
Same point with mysql:
1. logical shift
2. return 0 if c is less than zero

Difference with mysql:
1. sometimes retrun type of mysql bitshift will be converted to int128,

Consequence of difference:
1. `select BIGINT_MAX << 1` in mysql will return an int128 number say `18446744073709551614` we will not follow this behaviour

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/functions/function_bit_count.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    using ResultType = std::conditional_t<sizeof(T) * 8 >= 128, Int16, Int8>;
+
+    static inline ResultType apply(T a) {
+        if constexpr (std::is_same_v<T, Int128> || std::is_same_v<T, Int64> ||
+                      std::is_same_v<T, Int32> || std::is_same_v<T, Int16> ||
+                      std::is_same_v<T, Int8>) {
+            return std::popcount(static_cast<std::make_unsigned_t<T>>(a));
+        }
+
+        if constexpr (std::is_same_v<T, UInt128> || std::is_same_v<T, UInt64> ||
```
Comment (zhangstar333) [inline be/src/vec/functions/function_bit_count.cpp:49]: this types seems not support in doris?

File: docs/en/docs/sql-manual/sql-functions/bitwise-functions/bitshfitleft.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+{
+"title": "BIT_COUNT",
+"language": "en"
```
Comment (zhangstar333) [inline docs/en/docs/sql-manual/sql-functions/bitwise-functions/bitshfitleft.md:4]: this  title need change to bit_shift_left
and the zh-CN docs also not match

File: be/src/vec/functions/function_bit_count.cpp (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/functions/function_bit_count.cpp:49]: LAEGEINT from FE is regarded as Int128.

https://github.com/apache/doris/blob/b2a9c5b7d431f13fa28b0dc993f46a88430cfa46/be/src/vec/data_types/data_type_factory.cpp#L137

so we can apply bit_count on LARGEINT
```sql
select bit_count(cast (-1 as LARGEINT))
--------------

+---------------------------------+
| bit_count(cast(-1 as LARGEINT)) |
+---------------------------------+
|                             128 |
+---------------------------------+
```

File: docs/en/docs/sql-manual/sql-functions/bitwise-functions/bitshfitleft.md (hunk 2 above)
Comment (zhiqiang-hhhh) [reply docs/en/docs/sql-manual/sql-functions/bitwise-functions/bitshfitleft.md:4]: > this title need change to bit_shift_left and the zh-CN docs also not match

I will fix this

File: be/src/vec/functions/function_bit_count.cpp (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/functions/function_bit_count.cpp:49]: alright, maybe you mean unsigned data type is not supported? It's ok for me to remove the unsigned branch.

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/functions/function_bit_count.cpp (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/functions/function_bit_count.cpp]: removed

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Review by zhangstar333 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**